src/plugins/es_ui_shared/public/request/np_ready_request.ts

```diff
@@ -0,0 +1,166 @@
+/*
```

Comment:
This was largely borrowed from https://github.com/elastic/kibana/blob/master/src/plugins/es_ui_shared/public/request/request.ts, but updated to work with the NP http service.